### PR TITLE
[preparation 1/2]: refactor to simplify send logic, prepare for new RPCs

### PR DESCRIPTION
**Pull request by guggero**
This is a preparatory PR that does some of the refactoring required for https://github.com/lightninglabs/taproot-assets/pull/787. We also add two new fields to the vPSBT outputs: `ProofSuffix` and `ProofDeliveryAddress` which will be required to ship this missing information between different participants in a fully RPC driven send.

### CODE REVIEW

File: tappsbt/testdata/psbt_encoding_generated.json [hunk 1]
```diff
@@ -473,7 +476,8 @@
             "bip32_derivation": null,
             "tr_bip32_derivation": null,
             "tr_internal_key": "",
-            "tr_merkle_root": ""
+            "tr_merkle_root": "",
+            "proof_delivery_address": "https://example.com"
```
Comment (Roasbeef) [inline tappsbt/testdata/psbt_encoding_generated.json]: Add a hashmail test case as well?

File: tapfreighter/parcel.go [hunk 2]
```diff
@@ -300,10 +300,6 @@ type sendPackage struct {
 	// virtual asset transition transaction.
 	VirtualPacket *tappsbt.VPacket
 
-	// OutputIdxToAddr is a map from a VPacket's VOutput index to its
-	// associated Tap address.
-	OutputIdxToAddr tappsbt.OutputIdxToAddr
```
Comment (Roasbeef) [inline tapfreighter/parcel.go:305]: Nice simplification here enabled by just adding the courier addr to the output!

File: tappsbt/testdata/psbt_encoding_generated.json [hunk 3]
```diff
@@ -65,7 +65,7 @@
             "tr_bip32_derivation": null,
             "tr_internal_key": "",
             "tr_merkle_root": "",
-            "proof_delivery_address": "",
+            "proof_delivery_address": "hashmail://rand.hashmail.proof.courier:443",
```
Comment (Roasbeef) [inline tappsbt/testdata/psbt_encoding_generated.json:68]: scratch my other comment

**Review by Roasbeef — APPROVED** [review]
LGTM 🦄

**Review by GeorgeTsagk — APPROVED** [review]
LGTM + tACK :zap:

**Pull request merged**